astrowidgets/tests/widget_api_test.py

```diff
@@ -234,19 +241,58 @@ def test_add_markers(self):
         self.image.reset_markers()
         marknames = self.image._marktags
         assert len(marknames) == 0
-        assert self.image.get_markers(marker_name="all") is None
-        with pytest.warns(UserWarning, match='is empty'):
-            assert self.image.get_markers(marker_name=self.image._default_mark_tag_name) is None
+        self._check_marker_table_return_properties(self.image.get_markers(marker_name="all"))
+        # Check that no markers remain after clearing
+        tab = self.image.get_markers(marker_name=self.image._default_mark_tag_name)
+        self._check_marker_table_return_properties(tab)
+
+        # Check that retrieving a marker set that doesn't exist returns
+        # an empty table with the right columns
+        tab = self.image.get_markers(marker_name='test1')
+        self._check_marker_table_return_properties(tab)
+
+    def test_get_markers_accepts_list_of_names(self):
+        # Check that the get_markers method accepts a list of marker names
+        # and returns a table with all the markers from all the named sets.
+        data = np.arange(10).reshape((5, 2))
+        tab = Table(data=data, names=['x', 'y'])
+        self.image.add_markers(tab, marker_name='test1')
+        self.image.add_markers(tab, marker_name='test2')
 
-        with pytest.raises(ValueError, match="No markers named 'test1'"):
-            self.image.get_markers(marker_name='test1')
-        with pytest.raises(ValueError, match="No markers named 'test2'"):
-            self.image.get_markers(marker_name='test2')
+        # No guarantee markers will come back in the same order, so sort them.
+        t1 = self.image.get_markers(marker_name=['test1', 'test2'])
+        # Sort before comparing
+        t1.sort('x')
+        expected = vstack([tab, tab], join_type='exact')
+        expected.sort('x')
+        np.testing.assert_array_equal(t1['x'], expected['x'])
+        np.testing.assert_array_equal(t1['y'], expected['y'])
 
     def test_remove_markers(self):
         with pytest.raises(ValueError, match='arf'):
             self.image.remove_markers(marker_name='arf')
 
+    def test_remove_markers_name_all(self):
+        rng = np.random.default_rng(1234)
```

Comment:
There are remaining unnecessary random number generators that should be replaced with static arrays.